### PR TITLE
feat(RELEASE-1056): fix linting issues in check data key task

### PR DESCRIPTION
**Pull request by happybhati**
This commit fix the linting issue generated as checkton warnnig
 in the `check-data-key` task and test files.

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -53,19 +53,12 @@ jobs:
...
-            echo "Checkton found issues"
-            exit 1
-          else
-            echo "Checkton found no issues"
-          fi
+          find-copies-harder: false
+          # Set to false to disable the differential check
+          differential: false 
+          # Add a regex to include only the files you want to check
+          include-regex: "tasks/check-data-keys/.*\\.yaml"
```
Comment (mmalina) [inline .github/workflows/lint.yaml]: Opening a thread to make sure this is not left here. (Also, missing a newline at the end.)

File: tasks/check-data-keys/README.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ Currently, `releaseNotes`, and `cdn` are the only supported systems.
 | systems  | The systems to check that all data keys are present for | Yes      | []            |
 
 ## Changes in 0.9.1
-* The task would show failure output if a system with just one key was added (as cdn currently is)
+* Improved POSIX compliance and variable handling in the check-data-keys task.
```
Comment (mmalina) [inline tasks/check-data-keys/README.md]: This seems wrong. Should be a new changelog entry for the new version. Also, it should foremost say that we're fixing checkon/shellcheck issues.

**Review by mmalina — COMMENTED** [review]
I added two comments. Also, typo in commit message / PR title: warning.

Otherwise it looks good.

File: tasks/check-data-keys/README.md (hunk 2 above)
Comment (mmalina) [reply tasks/check-data-keys/README.md]: Also, why would we want to improve POSIX compliance? We're totally fine using bash and non-posix things.

**Review by johnbieren — COMMENTED** [review]
I agree with both of Martin's comments. Also can you fix the typo in the commit? It says `warnnig`

File: tasks/check-data-keys/README.md (hunk 2 above)
Comment (happybhati) [reply tasks/check-data-keys/README.md]: Yes, I agree. Fixed.
Yes, true. In many cases, it's better to use bash to adhere to our code style.

**Review by mmalina — APPROVED** [review]
LGTM. (But the language in the commit message is still slightly wrong.)

**Comment by happybhati** [comment]
> LGTM. (But the language in the commit message is still slightly wrong.)

Fixed thanks!

**Review by johnbieren — APPROVED** [review]

**Review by mmalina — APPROVED** [review]

**Pull request merged**